src/agoradatatools/etl/transform/gene_info.py

```diff
@@ -92,36 +93,32 @@ def transform_gene_info(
     # sort biodomains list alphabetically
     biodomains["biodomains"] = biodomains["biodomains"].apply(sorted)
 
+    # Type-check the 'is_adi' and 'is_tep' columns of tep_info to make sure they are booleans and not strings.
+    # Explicitly make NaN is_adi and is_tep values "False" to avoid having to check for boolean and NaN in the
+    # check below.
+    tep_info = tep_info.fillna({"is_adi": False, "is_tep": False})
+    if tep_info["is_adi"].dtype != bool:
+        raise TypeError("wrong data type in 'is_adi' column")
```

Comment:
NIT: You might want to make these error messages more specific

```suggestion
        raise TypeError(f"'is_adi' column must be `bool`, currently is {tep_info["is_adi"].dtype}")
```